tracks/swift/exercises/difference-of-squares/mentoring.md

```diff
@@ -0,0 +1,54 @@
+# Difference of Squares
+
+_Differences of Squares_ is a core exercise, unlocked after _Gigasecond_.
+
+## Reasonable Solutions
+
+```swift
+class Squares {
+    let maxNum: Int
+    
+    lazy var squareOfSum: Int = {
+        let sum = (0...self.maxNum).reduce(0, { $0 + $1 })
```

Comment:
I don't know Swift, so I take it there's no standard library `sum` method, since the "next level" solution defines it via `reduce`.

Is there a difference between generating a list and an iterable range between 0 and `self.maxNum`, and if so, do students sometimes pick the less optimal?

Reply:
There is no sum function within swift. The common practice is to do with with the reduce function.

Ranges and Lists are different types within swift, but each is essentially the same here because the methods used are defined on Collections which Ranges and Lists both are. I'll call that out here as well

Reply:
Excellent.